packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.tsx

```diff
@@ -143,34 +131,10 @@ ButtonUnstyled.propTypes /* remove-proptypes */ = {
    * @ignore
    */
   href: PropTypes.string,
-  /**
-   * @ignore
-   */
-  onBlur: PropTypes.func,
```

Comment:
@michaldudak I think we should go the other way around and explicitly define these props in the ButtonUnstyledProps interface, as if a custom component is used, that does not have these props in the props interface, there will be a typescript error, which would be a bug in my opinion.

Reply:
I'm not sure if I understand you correctly. Why would there be a TS error? We don't actually use these props anywhere. Also in ButtonUnstyled.spec.tsx we have a case with `CustomComponent` that don't have these defined and it works fine.

Reply:
We don't have with custom component. For e.g. this is failing:

```tsx
<ButtonUnstyled<typeof CustomComponent>
  slots={{ root: CustomComponent }}
  stringProp="test"
  numberProp={0}
  onClick={() => {}}
/>
```

Reply:
OK, I get it now. It's an interesting problem. The `ButtonUnstyled` doesn't handle (or require) the click (or several other) events, so strictly speaking, it's unnecessary. It's necessary in the `slots.root`, though, but there is no way of enforcing it.
In other words, even if we had `onClick` in `ButtonUnstyledProps`, it would be possible to provide `CustomComponent` that ignores its `onClick` prop and effectively create a broken button.

Reply:
Here is a failing test: https://app.circleci.com/pipelines/github/mui/material-ui/95773/workflows/1ec2c63d-7103-4340-b894-909bf873b7a0/jobs/509209 

I think we should update the types to add these event handlers.

Reply:
A couple of thoughts I have about this:

1. How do we decide which events to define explicitly?
2. If they are defined explicitly, we will lose polymorphism in their parameters. We'll have to type them as either `MouseEventHandler<Element>` or `MouseEventHandler<HTMLButtonElement>`. In either case, another test will fail (the last one from the .spec suite).
3. If we use this pattern, we should also update all the other components

Reply:
I see your point. I've added ts-expect-error directive on the new test, so that in the future, if we decide to handle this, we have some record that it was an intended decision. We can discuss it separately tough in the future.